app/controllers/bus_stops_controller.rb

```diff
@@ -81,7 +85,18 @@ def outdated
   def update
     @stop.assign_attributes stop_params
     @stop.decide_if_completed_by current_user
-    if @stop.save
+    if params[:commit] == 'Save and next'
+      route_id = session[:route_id]
```

Comment:
Session is like params - we should be prepared for values not to be there.

Reply:
* Particularly when we infer from the parameters that a particular session value will be present.